src/components/simulator/result/SimulatorProvider.tsx

```diff
@@ -134,20 +138,49 @@ export const SimulatorProvider: FC<SimulatorProviderProps> = ({ children }) => {
     handleAnimationStep();
   }, [handleAnimationStep]);
 
-  const pauseToggle = () => {
-    if (status.current === 'paused') {
-      status.current = 'running';
-      handleAnimationStep();
-    } else if (status.current === 'running') {
-      status.current = 'paused';
-    }
+  const pause = () => {
+    status.current = 'paused';
+    setAnimationData((prev) => [...prev]);
+  };
+
+  const unpause = () => {
+    status.current = 'running';
+    handleAnimationStep();
   };
 
   const end = () => {
     status.current = 'ended';
     setAnimationData(query.data?.data || []);
   };
 
+  const onBrush = (frame: number) => {
+    if (actionAfterBrushEnd.current === undefined) {
```

Comment:
super nitpick:
is there a specific reason to check undefined over `!actionAfterBrushEnd.current` ?

Reply:
fixed